pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/InMemoryDelayedDeliveryTracker.java

```diff
@@ -78,6 +83,17 @@ public boolean addMessage(long ledgerId, long entryId, long deliveryAt) {
             return false;
         }
 
+        synchronized (this) {
```

Comment:
I think we don't need `synchronized` here?

The `InMemoryDelayedDeliveryTracker` is not a thread-safe container.

Reply:
thanks